recirq/time_crystals/dtcexperiment.py

```diff
@@ -0,0 +1,144 @@
+# Copyright 2021 Google
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import cirq
+import itertools
+import numpy as np
+from typing import Sequence, Optional, Generator
+import os
+
+EXPERIMENT_NAME = "time_crystals"
+DEFAULT_BASE_DIR = os.path.expanduser(f'~/cirq_results/{EXPERIMENT_NAME}')
+
+class DTCExperiment:
+    """ Manage inputs to a DTC experiment, over some number of disorder instances
+
+    Attributes:
+        qubits: a chain of connected qubits available for the circuit
+        disorder_instances: number of disorder instances averaged over
+        initial_states: initial state of the system used in circuit
+        g: thermalization constant used in circuit
+        local_fields: random noise used in circuit
+        thetas: theta parameters for FSim Gate used in circuit
+        zetas: zeta parameters for FSim Gate used in circuit
+        chis: chi parameters for FSim Gate used in circuit
+        phis: phi parameters for FSim Gate used in circuit
+        gammas: gamma parameters for FSim Gate used in circuit
+
+    """
+
+    def __init__(
+            self,
+            qubits: Optional[Sequence[cirq.Qid]] = None,
+            disorder_instances: Optional[int] = 36,
+            g: Optional[int] = 0.94,
+            initial_states: Optional[np.ndarray] = None,
+            local_fields: Optional[np.ndarray] = None,
+            thetas: Optional[np.ndarray] = None,
+            zetas: Optional[np.ndarray] = None,
+            chis: Optional[np.ndarray] = None,
+            gammas: Optional[np.ndarray] = None,
+            phis: Optional[np.ndarray] = None
+            ):
+
+        self.qubits = qubits
+        self.disorder_instances = disorder_instances
+        self.g = g
+
+        if qubits is None:
+            qubit_locations = [(3, 9), (3, 8), (3, 7), (4, 7), (4, 8), (5, 8), (5, 7), (5, 6),
+                    (6, 6), (6, 5), (7, 5), (8, 5), (8, 4), (8, 3), (7, 3), (6, 3)]
+            self.qubits = [cirq.GridQubit(*idx) for idx in qubit_locations]
+        else:
+            self.qubits = qubits
+
+        num_qubits = len(self.qubits)
+
+        if initial_states is None:
+            self.initial_states = np.random.choice(2, (self.disorder_instances, num_qubits))
+        else:
+            self.initial_states = initial_states
+
+        if local_fields is None:
+            self.local_fields = np.random.uniform(-1.0, 1.0, (self.disorder_instances, num_qubits))
```

Comment:
you need to provide a textual description of these non-obvious default behaviors in the docstring (for this and other attributes)